File: python/src/animal.py
from datetime import datetime

class Animal:
    def __init__(self, animal_type, birth_date, arrival_date, weight, diseases=[], feeding_frequency="", feeding_type="", vaccinated=False):
        self._animal_type = animal_type
        self._birth_date = birth_date
        self._arrival_date = arrival_date
        self._weight = weight
        self._diseases = diseases
        self._feeding_frequency = feeding_frequency
        self._feeding_type = feeding_type
        self._vaccinated = vaccinated

    def get_animal_type(self):
        return self._animal_type

    def set_animal_type(self, value):
        self._animal_type = value

    def get_birth_date(self):
        return self._birth_date

    def set_birth_date(self, value):
        self._birth_date = value

    def get_arrival_date(self):
        return self._arrival_date

    def set_arrival_date(self, value):
        self._arrival_date = value

    def get_weight(self):
        return self._weight

    def set_weight(self, value):
        self._weight = value

    def get_diseases(self):
        return self._diseases

    def set_diseases(self, value):
        self._diseases = value

    def get_feeding_frequency(self):
        return self._feeding_frequency

    def set_feeding_frequency(self, value):
        self._feeding_frequency = value

    def get_feeding_type(self):
        return self._feeding_type

    def set_feeding_type(self, value):
        self._feeding_type = value

    def is_vaccinated(self):
        return self._vaccinated

    def set_vaccinated(self, value):
        self._vaccinated = value

File: src/Animal.java
import java.time.LocalDate;
import java.util.*;

public class Animal {
    static int nextId = 0;
    int id;
    String kind;
    LocalDate birthDate;
    LocalDate arrivalDate;
    float weight;
    ArrayList<String> diseases;
    String feedingFreq;
    String feedingType;
    boolean isVaccinated;
    
    // CONSTRUCTOR

    public Animal(String kind, LocalDate birthDate, LocalDate arrivalDate, float weight, ArrayList<String> diseases,
            String feedingFreq, String feedingType, boolean isVaccinated) {
        this.kind = kind;
        this.birthDate = birthDate;
        this.arrivalDate = arrivalDate;
        this.weight = weight;
        this.diseases = diseases;
        this.feedingFreq = feedingFreq;
        this.feedingType = feedingType;
        this.isVaccinated = isVaccinated;
        id = nextId;
        nextId++;
    }

    // GETTERS

    public int getID(){
        return id;
    }

    public float getWeight(){
        return weight;
    }

    public String getFeedingFreq(){
        return "Fed every " + feedingFreq;
    }

    public String getFeedingType(){
        return feedingType;
    }

    public String getVaccinationStatus(){
        if(isVaccinated){
            return "The animal is vaccinated";
        }else{
            return "The animal IS NOT vaccinated";
        }
    }

    public String getKind(){
        return kind;
    }

    public ArrayList<String> getDiseaseList(){
        return diseases;
    }

    public String getFeedingFrequency(){
        return feedingFreq;
    }

    public void getInfo(){
        System.out.println(" * ID: " + id);
        System.out.println(" * Kind: " + kind);
        System.out.println(" * Birth Date: " + birthDate.toString());
        System.out.println(" * Arrival Date: " + arrivalDate.toString());
        System.out.println(" * Weight: " + weight);

        if(diseases==null){
            System.out.println(" * No diseases");
        }else{
            System.out.println(" * Diseases:");
            for (int i = 0; i < diseases.size(); i++) {
                System.out.println("    * " + diseases.get(i));
            }
            System.out.println(" * Feeding Frequency: " + feedingFreq);
            System.out.println(" * Feeding Type: " + feedingType);
            if(isVaccinated){
                System.out.println(" * Vaccinated: Yes");
            }else{
                System.out.println(" * Vaccinated: No");
            }
        }
        
    }

    // SETTERS

    public void setWeight(float weight){
        this.weight = weight;
    }

    public void addDiseases(ArrayList<String> diseases){
        if(this.diseases!=null){
            this.diseases.addAll(diseases);
        }else{
            this.diseases = diseases;
        }
    }
    
    public void setFeedingFrequency(String freq){
        this.feedingFreq = freq;
    }

    public void setFeedingType(String type){
        this.feedingType = type;
    }

    public void setVaccineStatus(boolean status){
        this.isVaccinated = status;
    }

    
}
